.github/workflows/rust.yml

```diff
@@ -23,7 +23,7 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - run: rustup update
-    - run: cargo fmt --all --check
+    - run: make fmt
```

Comment:
Since there is no Cargo.toml in top level we would call make fmt instead to invoke Makefile in top level that recursively calls make fmt for sub-directories.

Reply:
Another option is to use `rustfmt **/*.rs` instead of `cargo fmt`. `cargo fmt` will format all rust files that are part of a project. `rustfmt` will format all rust files in any sub-directory. Remarkably this means `rustfmt` is often easier to use than the cargo equivalent.

Reply:
Thanks for the suggestion! I have made the change to use `rustfmt **/*.rs`.